### ISSUE TITLE
How should I set the learning rate on 8*v100 32g gpus？

### ISSUE THREAD

**Issue by zhangtao22** [Wangt-CN/DisCo#20]
This is the parameters I deployed
AZFUSE_USE_FUSE=0 QD_USE_LINEIDX_8B=0 NCCL_ASYNC_ERROR_HANDLING=0 mpirun -np 8 --allow-run-as-root python finetune_sdm_yaml.py --cf config/ref_attn_clip_combine_controlnet_attr_pretraining/coco_S256_xformers_tsv_strongrand.py --do_train --root_dir /data/DisCo --local_train_batch_size 64 --local_eval_batch_size 64 --log_dir exp/tiktok_pretrain --epochs 40 --deepspeed --eval_step 2000 --save_step 2000 --gradient_accumulate_steps 1 --learning_rate 1e-3 --fix_dist_seed --loss_target "noise" --train_yaml ./TSV_dataset/Human_Attribute_Pretrain/composite/train_TiktokDance-coco-single_person-Lindsey_0411_youtube-SHHQ-1.0-deepfashion2-laion_human-masks-single_cap.yaml --val_yaml ./TSV_dataset/Human_Attribute_Pretrain/composite/val_TiktokDance-coco-single_person-SHHQ-1.0-masks-single_cap.yaml --unet_unfreeze_type "transblocks" --refer_sdvae --ref_null_caption False --combine_clip_local --combine_use_mask --conds "masks" --max_eval_samples 2000 --strong_aug_stage1 --node_split_sampler 0
and the loss I got after Human attribute pretraining.
Metering:{'loss_total': '0.0667'}: 100%|██████████| 55280/55280 [81:48:04<00:00,  5.33s/it]
![1689906765826](https://github.com/Wangt-CN/DisCo/assets/35836562/37e7de7f-3085-474a-9a11-22dcdeefbd6c)

I noticed that on your article, you mentioned All pre-training experiments are conducted on 4x8 NVIDIA V100 GPUs for 25K steps with image size 256×256 and learning rate 1e−3. 
Because I only got one-fourth of the number of gpus you have，should I reduce the learning rate to one-fourth of 1e-3？
what is your loss after training on this stage？Thank you！

**Comment by Wangt-CN** [comment]
Hi, I have tried to multiple the learning rate by the rate that the batch size increase, but I am not sure if it is the best lr under that situation. For loss, different setting causes different loss values, I recommend you to check the final eval FID after training.

**Comment by zhangtao22** [comment]
Ultimately,the indicators I got are much worse than yours
{'L1': 0.00039951090145022176, 'SSIM': 0.657848540669247, 'LPIPS': 0.2970971225998282, 'PSNR': 28.89969279008445, 'clean-fid': 32.09775464320444}
FID:  31.576945674116274

**Comment by Wangt-CN** [comment]
> Hi, I have tried to multiple the learning rate by the rate that the batch size increase, but I am not sure if it is the best lr under that situation. For loss, different setting causes different loss values, I recommend you to check the final eval FID after training.

This is my experience for the 2nd stage (fine-tuning). For pre-training, we do not try to use only 1x8 V100 ever. Maybe you should try different lr, but I recommend you not to re-train from scratch but use our checkpoint as the starting point. The FID of 31 is a little bit high. At least it should under 25.
